File: entrypoint/src/main/kotlin/api/EditorController.java
package api;

import org.springframework.web.bind.annotation.*;
import purple.lemon.dashboard.model.EditorData;
import purple.lemon.dashboard.model.EditorDataVO;
import purple.lemon.dashboard.repository.EditorRepository;

import java.util.Optional;

@CrossOrigin(origins = {"http://localhost:1994"})
@RestController
@RequestMapping("/api/editor")
public class EditorController {

    final EditorRepository editorRepository;

    public EditorController(EditorRepository editorRepository) {
        this.editorRepository = editorRepository;
    }

    @GetMapping("/content")
    public EditorDataVO getEditorData() {
        Optional<EditorData> optionalEditorData = editorRepository.getEditorData(1);
        return optionalEditorData.map(EditorDataVO::transform).orElseGet(EditorDataVO::empty);
    }

    @PutMapping("/store")
    public void updateEditorData(@RequestParam(value = "dataToStore") String dataToStore) {
        if (dataToStore != null) {
            editorRepository.saveDataContent(1, dataToStore);
        }
    }
}

File: application/entrypoint/src/main/kotlin/purple/lemon/controller/EditorController.kt
package purple.lemon.controller

import org.springframework.web.bind.annotation.*
import purple.lemon.controller.dto.EditorDataDto
import purple.lemon.usecase.editor.EditorUseCase

@CrossOrigin(origins = ["http://localhost:1994"])
@RestController
@RequestMapping("/api/editor")
class EditorController(
        private val editorUseCase: EditorUseCase
) {

    @GetMapping("/content")
    fun getEditorData(): EditorDataDto {
        val foundEditorData = editorUseCase.getEditorData(EditorUseCase.Request(1))
        return EditorDataDto.transform(foundEditorData.editorData)
    }

    @PutMapping("/store")
    fun updateEditorData(@RequestParam(value = "dataToStore") dataToStore: String) {
        if (dataToStore != null || dataToStore != "") {
            editorUseCase.saveEditorData(EditorUseCase.Request(1, dataToStore))
        }
    }
}